src/ast/CMakeLists.txt

```diff
@@ -39,8 +71,27 @@ if (STATIC_LINKING)
       clangSema
       clangSerialization
       clangToolingCore)
-  llvm_map_components_to_libnames(llvm_libs bpfcodegen ipo irreader mcjit orcjit ${LLVM_TARGETS_TO_BUILD})
-  target_link_libraries(ast ${clang_libs})
+  llvm_map_components_to_libnames(llvm_libs bpfcodegen ipo irreader mcjit orcjit option ${LLVM_TARGETS_TO_BUILD})
+
+  # Workaround to ensure that the LLVM DSO is not linked
+  function(unlink_transitive_dependency targets dep_to_remove)
+    foreach(tgt ${targets})
+      get_target_property(int_link_libs ${tgt} INTERFACE_LINK_LIBRARIES)
+
+      if(NOT(int_link_libs MATCHES "-NOTFOUND$"))
+        list(REMOVE_ITEM int_link_libs "${dep_to_remove}")
+        list(JOIN int_link_libs ";" sanitized_link_libs)
```

Comment:
Joining a semicolon isn't necessary on CMake, and the JOIN primitive is actually new for lists in cmake, so this function doesn't work on the min supported CMake version (curently claiming to be `cmake_minimum_required(VERSION 2.8.12)` but i'm not sure I believe that). In any case, this doesn't work with the Debian default cmake version of 3.10).